### ISSUE TITLE
Question about flow loss

### ISSUE THREAD

**Issue by moordo91** [YoushaaMurhij/OFMPNet#4]
I have a question about your implementation at OFMPNet/core/losses /loss.py

```python
  def _flow_loss(
      self,
      true_flow: torch.Tensor,
      pred_flow: torch.Tensor,
      loss_weight: float = 1,
  ) -> torch.Tensor:
      """Computes L1 flow loss."""
      diff = true_flow - pred_flow
      # Ignore predictions in areas where ground-truth flow is zero.
      # [batch_size, height, width, 1], [batch_size, height, width, 1]
      true_flow_dx, true_flow_dy = torch.chunk(true_flow, 2, dim=-1)

      # [batch_size, height, width, 1]
      flow_exists = torch.logical_or(
          torch.not_equal(true_flow_dx, 0.0),
          torch.not_equal(true_flow_dy, 0.0),
      )
      flow_exists = flow_exists.to(torch.float32)
      diff = diff * flow_exists
      diff_norm = torch.linalg.norm(diff, ord=1, dim=-1)  # L1 norm.
      diff_norm_sum = torch.sum(diff_norm)
      flow_exists_sum = torch.sum(flow_exists)*self.replica / 2 # / 2 since (dx, dy) is counted twice.
      if torch.is_nonzero(flow_exists_sum):
          mean_diff = torch.div(diff_norm_sum, flow_exists_sum)
      else:
          mean_diff = 0
      return loss_weight * mean_diff
```

`flow_exists_sum = torch.sum(flow_exists)*self.replica / 2 `

I think you shouldn't devide flow_exists_sum by 2, since (dx, dy) is not counted twice.
You get it from torch.logical_or() so it is not counted twice.

Am I wrong? then please correct me.

**Comment by YoushaaMurhij** [comment]
Yeah, that's correct! But this should not affect the overall performance!
I actually got this loss from [strajnet](https://github.com/georgeliu233/STrajNet/blob/21884df0ea34c725f970c26267eb191fa3c9e8f6/loss.py#L294)

**Comment by MPCheng-ZW** [comment]
Why am I getting such high flow loss?

<img width="626" alt="微信图片_20241219225337" src="https://github.com/user-attachments/assets/3f6a7e58-d470-4d98-8300-054ac663ed00" />

Is this correct?

Thanks!

@YoushaaMurhij @moordo91

**Comment by YoushaaMurhij** [comment]
> Why am I getting such high flow loss?
> 
> <img alt="微信图片_20241219225337" width="626" src="https://private-user-images.githubusercontent.com/73774196/397421831-3f6a7e58-d470-4d98-8300-054ac663ed00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQ2MjI1NzQsIm5iZiI6MTczNDYyMjI3NCwicGF0aCI6Ii83Mzc3NDE5Ni8zOTc0MjE4MzEtM2Y2YTdlNTgtZDQ3MC00ZDk4LTgzMDAtMDU0YWM2NjNlZDAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjE5VDE1MzExNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWU2MjQxZWMyZDA3YTkyMmYzNTVmMTJlMzcyNGZhMDhkM2E4NGM3OGJkMTllMzQ1MThlMTUxZjk3Njc3NTdhMzQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.ZTTIaUZAxsUetgQ56PKmKZ2YM7MTaNj_Efe-Oy36lS0">
> Is this correct?
> 
> Thanks!
> 
> @YoushaaMurhij @moordo91

I think it is normal if this is the very start of training process! I recommend to increase batch size for a stable training process

**Comment by MPCheng-ZW** [comment]
What is the approximate value of the final flow loss after 15 epochs of training?

How many seconds does it take to train a sample? ( I mean xx s/it)

Thanks!

**Comment by YoushaaMurhij** [comment]
> What is the approximate value of the final flow loss after 15 epochs of training?
> 
> How many seconds does it take to train a sample? ( I mean xx s/it)
> 
> Thanks!

Final loss was between 50 - 70. Sorry, I kept only the graphs. I dont have xx s/it info anymore!
